src/connectors/impls/kafka/consumer.rs

```diff
@@ -28,43 +30,184 @@ use indexmap::IndexMap;
 use log::Level::Debug;
 use rdkafka::config::ClientConfig;
 use rdkafka::consumer::{CommitMode, Consumer, ConsumerContext, Rebalance, StreamConsumer};
-use rdkafka::error::KafkaError;
+use rdkafka::error::{KafkaError, KafkaResult};
 use rdkafka::message::{BorrowedMessage, Headers, Message};
 use rdkafka::{Offset, TopicPartitionList};
 use rdkafka_sys::RDKafkaErrorCode;
 
 const KAFKA_CONSUMER_META_KEY: &str = "kafka_consumer";
 
+#[derive(Deserialize, Debug, Clone, Default)]
+#[serde(deny_unknown_fields, rename_all = "lowercase")]
+enum Mode {
+    #[default]
+    /// - enable.auto.commit: true,
+    /// - enable.auto.offset.store: true,
+    /// - auto.commit.interval.ms: 5000 # 5s
+    Performance,
+    /// - enable.auto.commit: true
+    /// - enable.auto.offset.store: false
+    /// - auto.commit.interval.ms: # take the configured value
+    /// - retry_failed_events: true
+    Transactional {
+        #[serde(default = "default_commit_interval")]
+        commit_interval: u64,
+    }, // enable.auto.commit: true, enable.auto.offset.store: false, retry_failed_events: true
+    Custom {
+        /// Optional rdkafka configuration
+        ///
+        /// Can be overwritten by tremor for settings required for `auto_commit` and/or `retry_failed_events`.
+        rdkafka_options: HashMap<String, String>,
+
+        /// This config determines the behaviour of this source. DANGER: this might lead to a lot of traffic to the group-coordinator
+        /// if `enable.auto.commit` is set to `false`:
+        ///
+        /// if set to `true` this source will reset the consumer offset to a
+        /// failed message, so it will effectively retry those messages.
+        ///
+        /// If set to `false` this source will only commit the consumer offset
+        /// if the message has been successfully acknowledged.
+        ///
+        /// This might lead to events being sent multiple times.
+        /// This should not be used when persistent errors are expected (e.g. if the message content is malformed and will lead to repeated errors)
```

Comment:
```suggestion
        /// This config determines the behaviour of this source.
        ///
        /// If `enable.auto.commit` is set to `false`:
        ///
        /// If `retry_failed_events` is set to `true` this source will reset the consumer offset to a
        /// failed message, so it will effectively retry those messages.
        /// DANGER: this might lead to a lot of traffic to the group-coordinator!
        ///
        /// If `retry_failed_events` is set to `false` this source will only commit the consumer offset
        /// if the message has been successfully acknowledged.
        ///
        /// This might lead to events being sent multiple times.
        /// This should not be used when persistent errors are expected (e.g. if the message content is malformed and will lead to repeated errors)
```